pkg/waiter/waiter_test.go

```diff
@@ -0,0 +1,197 @@
+package waiter
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/stackrox/rox/pkg/sync"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestStringWaiter(t *testing.T) {
+	wm := NewManager[string]()
+	wm.Start(context.Background())
+
+	w, err := wm.NewWaiter()
+	assert.NoError(t, err)
+
+	want := "hello"
+	require.NoError(t, wm.Send(w.ID(), want, nil))
+
+	got, err := w.Wait(context.Background())
+	assert.NoError(t, err)
+	assert.Equal(t, got, want)
+}
+
+func TestStructWaiter(t *testing.T) {
+	type widget struct {
+		msg string
+	}
+
+	wm := NewManager[widget]()
+	wm.Start(context.Background())
+
+	w, err := wm.NewWaiter()
+	assert.NoError(t, err)
+
+	want := widget{msg: "hello"}
+	require.NoError(t, wm.Send(w.ID(), want, nil))
+
+	got, err := w.Wait(context.Background())
+	assert.NoError(t, err)
+	assert.Equal(t, got.msg, want.msg)
+}
+
+func TestPointerWaiter(t *testing.T) {
+	type widget struct {
+		msg string
+	}
+
+	wm := NewManager[*widget]()
+	wm.Start(context.Background())
+
+	w, err := wm.NewWaiter()
+	assert.NoError(t, err)
+
+	want := &widget{msg: "hello"}
+	require.NoError(t, wm.Send(w.ID(), want, nil))
+
+	got, err := w.Wait(context.Background())
+	assert.NoError(t, err)
+	assert.Equal(t, got, want)
+	if got != want {
```

Comment:
isnt this already covered by `assert.Equal`?

Reply:
Yup! removed.